packages/cli/lib/projects.js

```diff
@@ -699,9 +699,51 @@ const makePollTaskStatusFunc = ({
 
     return new Promise((resolve, reject) => {
       const pollInterval = setInterval(async () => {
-        const taskStatus = await statusFn(accountId, taskName, taskId).catch(
-          reject
-        );
+        let taskStatus;
+        try {
+          taskStatus = await statusFn(accountId, taskName, taskId);
+        } catch (e) {
+          logApiErrorInstance(
+            e,
+            new ApiErrorContext({
+              accountId,
+              projectName: taskName,
+            })
+          );
+          return reject(
```

Comment:
I'm curious how this error handling will impact the downstream code. Do we ever try to catch the errors thrown by this build/deploy polling util?

Reply:
Great question.  I was assuming that since this method was calling `reject` previously that it should be handled, but that wasn't always the case.  I'm going to push up a couple places that aren't handling the rejection with some TODO comments and address.

Reply:
> I'm curious how this error handling will impact the downstream code. Do we ever try to catch the errors thrown by this build/deploy polling util?


It looks like we don't explicitly catch the errors off of any of the polling methods.  I added that commit with the todos, but ended up reverting it because I couldn't think of anything else to do in those situations other than log, and I centralized that in this method.